dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/SdkFeature.kt

```diff
@@ -166,6 +170,39 @@ internal class SdkFeature(
 
     // endregion
 
+    // region Context Update Listener
+
+    internal fun notifyContextUpdated(context: Map<String, Any?>) {
+        contextUpdateListeners.keys.forEach {
+            it.onContextUpdate(context)
+        }
+    }
+
+    internal fun setContextUpdateListener(listener: FeatureContextUpdateListener) {
+        synchronized(contextUpdateListeners) {
```

Comment:
Do we actually need `ConcurrentHashMap` then if each access that modifies the state of `contextUpdateListeners` is backed by `synchronized`?

Reply:
yes we need it as I did not want to `synchronize` when we notify the listeners. When I travers them for notification another thread might remove or add a new one and it will get into `ConcurrentModificationException`